### PR TITLE
Prince/ Add trust pilot to homepage

### PR DESCRIPTION
**Pull request by prince-deriv**
Changes:
 - updated `what our clients say` section
 - added trustpilot API to graphql
 - added tab component and trustpilot component

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [x] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: .gitignore
Comment (habib-binary) [inline .gitignore:56]: Hi Prince, let's add .env.development instead of Asterix

File: src/pages/home/_trust_pilot.js
Comment (habib-binary) [inline src/pages/home/_trust_pilot.js:130]: Let's add alt tag

File: src/pages/home/_trust_pilot.js
Comment (habib-binary) [inline src/pages/home/_trust_pilot.js:138]: Pleas add the calculation in a variable and just use it in the component

File: src/pages/home/_trust_pilot.js
Comment (habib-binary) [inline src/pages/home/_trust_pilot.js:146]: Let's do the logical part using a constant and use it here

File: src/pages/home/_what-our-clients-say.js
Comment (habib-binary) [inline src/pages/home/_what-our-clients-say.js:136]: Add alt tag for all images

**Review by habib-binary — COMMENTED** [review]
A few minor issues

File: src/pages/home/_trust_pilot.js
Comment (sean-binary) [inline src/pages/home/_trust_pilot.js:104]: Prefixing it with `GATSBY_` will make it available in the browser. You should remove it for this use case.

File: src/pages/home/_what-our-clients-say.js
Comment (sean-binary) [reply src/pages/home/_what-our-clients-say.js]: Note that only images with no supporting text will need alts with descriptions. For icons that act as supplementary elements to beautify the section, those will need empty alt tags.

**Comment by sean-binary** [comment]
Please also take note of the code smell that is detected and try to address that 🙏

File: gatsby-browser.js
Comment (sean-binary) [inline gatsby-browser.js:164]:
```suggestion
    addScript({
        src: 'https://widget.trustpilot.com/bootstrap/v5/tp.widget.bootstrap.min.js',
        id: 'trust-pilot',
        async: true
    })
```

Comment (sean-binary) [reply gatsby-browser.js:164]: Let's make this async.

File: src/images/svg/playstore/stars/index.js
Comment (habib-binary) [inline src/images/svg/playstore/stars/index.js:1]: Hi @prince-deriv , do we still need this, as we're using the widget now?

**Comment by sean-binary** [comment]
There's also a nested ternary code smell flagged by sonarcloud, could you take a look at it?

**Review by habib-binary — DISMISSED** [review]
LGTM ~

**Comment by sean-binary** [comment]
The appstore and the google play store SVGs seem to be very large SVGs. It's probably more ideal to use the PNG formats for those otherwise you might get console warnings about it.

File: src/images/svg/playstore/stars/index.js
Comment (sean-binary) [reply src/images/svg/playstore/stars/index.js:1]: @prince-deriv Are we removing these star svgs and commenting out the logic for now?

File: src/pages/home/_what-our-clients-say.js
Comment (sean-binary) [inline src/pages/home/_what-our-clients-say.js:93]: Should be able to pass as props.

File: src/pages/home/_what-our-clients-say.js
Comment (sean-binary) [inline src/pages/home/_what-our-clients-say.js:187]: Let's store all the static links in constants.js

File: src/components/custom/_line-tab.js
Comment (sean-binary) [inline src/components/custom/_line-tab.js:11]: You can pass all these as props to `Flex`.

File: src/pages/home/_what-our-clients-say.js
Comment (sean-binary) [inline src/pages/home/_what-our-clients-say.js:128]: It would be a bit more helpful if these variables have more meaningful names 😅

File: src/pages/home/_what-our-clients-say.js
Comment (sean-binary) [inline src/pages/home/_what-our-clients-say.js:109]: Same here.

File: src/pages/home/_what-our-clients-say.js
Comment (prince-deriv) [reply src/pages/home/_what-our-clients-say.js:51]: No @sean-binary  it's targeting our elements to align the widget properly.

File: src/images/svg/playstore/stars/index.js
Comment (prince-deriv) [reply src/images/svg/playstore/stars/index.js:1]: @sean-binary  this is the small black stars for Play Store review. It's a ready component waiting for dynamic values from API later

File: src/common/utility.js
Comment (mahdiaryayi-fs) [inline src/common/utility.js:255]: Can we apply the settings in one step like this?

```suggestion
    let script = document.createElement('script');
    
    script = {
        ...script,
        ...settings,
    }
```

Comment (prince-deriv) [reply src/common/utility.js:255]: Yes @mahdiaryayi-fs  that would be cleaner

Comment (prince-deriv) [reply src/common/utility.js]: Hi @mahdiaryayi-fs  found out that the script is a`DOM` element and not an object we can just merge by spreading. I did another work around instead.

**Review by mahdiaryayi-fs — APPROVED** [review]

**Review by sean-binary — DISMISSED** [review]
LGTM 👍  Just a minor improvement that can be worked on moving forward. You can pass in styles for media queries to any component that extends `Box` which is usually `Flex` or `SectionContainer` which accepts a style object into props like `tabletL`, `mobileL` etc.

File: src/components/custom/_line-tab.js
Comment (mustofa-binary) [inline src/components/custom/_line-tab.js:1]: i dont see this _line tab component used anywhere, may i know where are we using this?

**Review by mustofa-binary — CHANGES_REQUESTED** [review]

File: src/components/custom/_line-tab.js
Comment (prince-deriv) [reply src/components/custom/_line-tab.js:1]: @mustofa-binary we are using it in the new trustpilot section. It is planned to have 3 different tabs in the future until approval. For now it is there but won't show up any clickable navigations coz it is just 1 item.

File: gatsby-browser.js
Comment (mahdiaryayi-fs) [reply gatsby-browser.js:27]: @mustofa-binary I get your point but generally, the `for..in` loop is better to be avoided as much as possible. because `for..in` loops not only iterate over keys of the object but also iterate over the prototype chain that may create bugs if is not intended. but `Object.keys()` is only extract the own keys of the object and does not go through prototype chain
here is an example:

```js
const parent = {
  name: 'dad',
};

const son = Object.create(parent);

son.age = 20;

for (let key in son) {
  console.log(key); // prints both "age" and "name"
}

console.log(Object.keys(son)); // prints only ["age"]
```

File: .gitignore
Comment (prince-deriv) [reply .gitignore:56]: For now we can omit this. We are not using any env variables after `DevOps` refused to let us use them as it will be expose in the browser.

**Review by mahdiaryayi-fs — APPROVED** [review]

File: gatsby-browser.js
Comment (mustofa-binary) [reply gatsby-browser.js]: actually the for in seems to be more as expected based on your example, since we expect son to hv name and age as well inherit from parent. but, here we want settings object not to be altered at all cost, so yes, Object.keys will do the job, fair point, thank you @mahdiaryayi-fs

File: src/common/utility.js
Comment (mustofa-binary) [reply src/common/utility.js:248]: @prince-deriv ?

File: src/pages/home/_what-our-clients-say.js
Comment (mustofa-binary) [inline src/pages/home/_what-our-clients-say.js:137]:
```suggestion
    const rating_str = `${rating}`
```

File: src/pages/home/_what-our-clients-say.js
Comment (mustofa-binary) [inline src/pages/home/_what-our-clients-say.js:141]:
```suggestion
        if (decimal_place) {
```
truthy check

File: src/pages/home/_what-our-clients-say.js
Comment (mustofa-binary) [inline src/pages/home/_what-our-clients-say.js:258]: alt?

**Review by mustofa-binary — COMMENTED** [review]
nit: everything else lgtm

**Review by mustofa-binary — DISMISSED** [review]

**Review by habib-binary — APPROVED** [review]
LGTM

**Pull request merged**
